src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,129 @@
+        if (!directory.exists()) {
+            System.out.println("'data' directory not found. Creating the directory...");
+            directory.mkdir();
+            System.out.println("Directory created.");
+        }
+
+        Path filepath = Paths.get(home, "ip", "data", "duke.txt");
+        File file = new File(filepath.toString());
+        if (!file.exists()) {
+            System.out.println("Data file not found. Creating a new file...");
+            try {
+                file.createNewFile();
+                System.out.println("File created.");
+            } catch (IOException e) {
+                System.out.println("IO error occured");
+            }
+        }
+        return file;
+    }
+
+    /**
+     * Writes the saved tasks from previous sessions (if any) into
+     * the local TaskList object.
+     *
+     * @return An ArrayList initialised with the previously saved tasks.
+     * @throws DukeException If an IOException occurs upon reading of file, which is passed on to
+     *                       the main() method of Duke class.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        try {
+            file = this.getfiles();
+            FileReader reader = new FileReader(file);
+            BufferedReader buffreader = new BufferedReader(reader);
+            String text;
+            ArrayList<Task> saved = new ArrayList<>();
+            while ((text = buffreader.readLine()) != null) {
+                String[] lineArr = text.split(" | ");
+                switch (lineArr.length) {
+                case 3:
+                    // To complete TaskList class
+                    Task toDo = new Task(lineArr[2]);
+                    saved.add(toDo);
+                    if (lineArr[1].equals("1")) {
+                        toDo.complete();
+                    }
+                    break;
+                case 4:
+                    if (lineArr[0].equals("D")) {
+                        Deadline deadline = new Deadline(lineArr[2], LocalDateTime.parse(lineArr[3],
+                                this.dateTimeFormat));
+                        saved.add(deadline);
+                    } else if (lineArr[0].equals("E")) {
+                        Event event = new Event(lineArr[2], LocalDateTime.parse(lineArr[3], this.dateTimeFormat));
+                        saved.add(event);
+                    }
+                    break;
+                default:
+                    throw new DukeException("Error loading input file.");
+                }
+            }
+            return saved;
+        } catch (IOException e) {
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+    /**
+     * Writes the tasks of the current session into the dynamically created file
+     * on the local filesystem.
+     *
+     * @throws DukeException If an IOException occurs during writing of file, which
+     *                       is passed on to the main() method of Duke class.
+     */
```

Comment:
```suggestion
    /**
     * Writes the tasks of the current session into the dynamically created file
     * on the local filesystem.
     * 
     * @param list ...
     *
     * @throws DukeException If an IOException occurs during writing of file, which
     *                       is passed on to the main() method of Duke class.
     */
```

Would it be a good idea to add Javadoc comments for method parameters as well? Might help fellow developers understand what the inputs to your methods are. The form of Javadoc comments is specified in the [Comments](https://se-education.org/guides/conventions/java/intermediate.html#comments) section of the course coding standard.